### ISSUE TITLE
"There was an error fetching the favorite status: Internal Server Error" flash message for anonymous users

### ISSUE THREAD

**Issue by trepmag** [apache/superset#14070]
Anonymous users get the following flash message while browsing chars or dashboards lists: "There was an error fetching the favorite status: Internal Server Error".

Also, anonymous users are presented with favorites switches in those lists and also in the chart explorer header while in the dashboard header it is correctly hided. 

### Expected results

As an anonymous user the above described flash message and favorites switch shouldn't appear!...

### Actual results

As an anonymous user the above described flash message and favorites switch appear!...

#### Screenshots

![image](https://user-images.githubusercontent.com/875021/114268387-11403f00-9a01-11eb-9f6e-8f76b298cc95.png)

#### How to reproduce the bug

1. Set some Charts or Dashboards to be available to Public
1. As an anonymnous user, go to http://localhost:9000/chart/list/ or http://localhost:9000/dashboard/list/
1. Wait page load
1. See the above described flash message and the favorites switches

### Environment
- superset version: `1.0.1` (git)
- python version: `3.7.9` (docker env)
- node.js version: `12` (docker env)

#### Browser
```
GET http://localhost:9000/api/v1/dashboard/favorite_status/?q=%21%284%2C2%2C1%29 Status: 500
INTERNAL SERVER ERROR
```

#### Superset app
```
superset_app             | ERROR:root:'AnonymousUserMixin' object has no attribute 'id'
superset_app             | Traceback (most recent call last):
superset_app             |   File "/usr/local/lib/python3.7/site-packages/flask_appbuilder/api/__init__.py", line 84, in wraps
superset_app             |     return f(self, *args, **kwargs)
superset_app             |   File "/app/superset/views/base_api.py", line 79, in wraps
superset_app             |     duration, response = time_function(f, self, *args, **kwargs)
superset_app             |   File "/app/superset/utils/core.py", line 1289, in time_function
superset_app             |     response = func(*args, **kwargs)
superset_app             |   File "/usr/local/lib/python3.7/site-packages/flask_appbuilder/api/__init__.py", line 154, in wraps
superset_app             |     return f(self, *args, **kwargs)
superset_app             |   File "/app/superset/utils/log.py", line 164, in wrapper
superset_app             |     value = f(*args, **kwargs)
superset_app             |   File "/app/superset/dashboards/api.py", line 662, in favorite_status
superset_app             |     favorited_dashboard_ids = DashboardDAO.favorited_ids(dashboards, g.user.id)
superset_app             |   File "/usr/local/lib/python3.7/site-packages/werkzeug/local.py", line 347, in __getattr__
superset_app             |     return getattr(self._get_current_object(), name)
superset_app             | AttributeError: 'AnonymousUserMixin' object has no attribute 'id'
superset_app             | 192.168.112.1 - - [10/Apr/2021 11:36:59] "GET /api/v1/dashboard/favorite_status/?q=%21%284%2C2%2C1%29 HTTP/1.1" 500 -
superset_app             | INFO:werkzeug:192.168.112.1 - - [10/Apr/2021 11:36:59] "GET /api/v1/dashboard/favorite_status/?q=%21%284%2C2%2C1%29 HTTP/1.1" 500 -
```

**Comment by squalou** [comment]
Hi,

on my side, with 1.0.1 and 1.1.0-rc2 I'm getting this popup

```There was an error fetching the favorite status: NetworkError when attempting to fetch resource.```

*even* with logged in users. Did you notice something like that ?

**Comment by amitmiran137** [comment]
waiting for the PR ;)

**Comment by trepmag** [comment]
Lint commit 43fe481 imposed lots of diff ?!?!...

**Comment by amitmiran137** [comment]
Could you open a PR?

**Comment by trepmag** [comment]
Fix into 6875a1a (#14287)...
